apps/desktop/src/components/v3/UnifiedDiffView.svelte

```diff
@@ -99,22 +89,26 @@
 	<ReduxResult result={diffQuery}>
 		{#snippet children(diff)}
 			{#if diff.type === 'Patch'}
-				{#each diff.subject.hunks as hunk}
-					<HunkDiff
-						filePath={path}
-						hunkStr={hunk.diff}
-						selected={selectable
-							? selection &&
-								(selection.type === 'full' ||
-									selection.hunks.some((h) => h.newStart === hunk.newStart))
-								? true
-								: false
-							: undefined}
-						onchange={(selected) => {
-							onchange(hunk, selected, diff.subject.hunks);
-						}}
-					/>
-				{/each}
+				{#if diff.subject.hunks.length}
```

Comment:
I think you can drop this if statement? Iterating of a zero length array is a no-op.

Reply:
Hmm yeah logically you're probably right. I had it here in order to be able ot render at least something (in this case the string "No Content"), when there are no hunks.

Unfortunately the way we have it architected now, we don't know the contents of the file change, i.e. the diff, until when we open it. So its not straight forward to just disable toggling open of a file without checking them all beforehand, which is what we wanted to avoid, right.

Reply:
Good point.

@krlvi I think in v2 code we generate a synthetic hunk when there is no diff. It's easier for the front end if we keep that behaviour, but what do you think?

<img width="529" alt="image" src="https://github.com/user-attachments/assets/e771470c-d309-449a-bbb1-535e303cf37c" />

Reply:
Surely we could add a synthetic hunk, but on face value this feels like masking another problem. It's totally valid to have no hunks in the file (eg. file executable bit added/removed) - and the app could/should show that.
At the moment I feel like I'd rather favor correctness (of the API) over supporting the existing semantic.

Is there something that I might be missing here?

Reply:
And I guess the same applies for commits with no files in them - the app allows their creation, and it should be still possible to preview / edit the message

Reply:
No you're not missing anything, it's indeed more correct. If we had a synthetic hunk, it also take extra handling in dragging that to a commit. But, we do need think about how to render this. @PavelLaptev any chance you could help out here?

Currently it looks like this when you preview a new file with no content.
<img width="458" alt="image" src="https://github.com/user-attachments/assets/640bae4b-150c-45e2-b6cf-eb56f526924a" />

Reply:
@mtsgrd thanks, I'll take a look today 👀

Reply:
This might be the time to create a view for how files look like when they have their flags changed, or if a file became a symlink etc

Reply:
@krlvi got it. But it's better to discuss the design task for in a separate thread or issue

Reply:
@krlvi Yeah +1

I was just reading through the types again and it seems like we could disable the opening of these types of file with those flags too. Would it be a lot of work on your end to add a  `EmptyFile` `TreeChange` `subject` flag? 

Also I haven't seen any of them on the FE yet. Are they maybe just not implemented yet?

Reply:
@ndom91 my intuition is that we should still allow the file to be opened, but show a message where we otherwise have a diff.

Reply:
Okay yeah I'm open to whatever. Maybe Pavel has a nice design idea haha

Reply:
Not yet, but we will at some point. For now we can just drop some text in its place?

Reply:
Right. Currently there is some text there and it just says "No Content" i.e.:

![image](https://github.com/user-attachments/assets/a28d7db8-76aa-47b6-8b00-b794cd1f4c83)